drivers/pcie/host/pcie.c

```diff
@@ -5,6 +5,14 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#if defined(CONFIG_PCIE_PRT)
+#if defined(CONFIG_X86_64)
+#define DT_DRV_COMPAT intel_pcie
```

Comment:
It looks like at least arm64 qemu is using `drivers/pcie/host/pcie.c`, however it's using `compatible = "pci-host-ecam-generic"` for its PCIe bus node. This would lead to the conclusion, that `drivers/pcie/host/pcie.c` is not a driver in itself, rather a kind of "generic PCIe code" that can be used by specific PCIe bus drivers? If that's the correct assessment, and we don't want to change this design, then anything that is  arch-specific should be moved into an arch-specific c-file, including all `DT_*()` macro invocations.

Reply:
Think is, the dts bindings for intel's pcie has nothing different from pci. In fact, I don't see why intel's pcie controller would have anything specific. 

Anyway, make it  #define DT_DRV_COMPAT pcie_controller

Reply:
> Anyway, make it  #define DT_DRV_COMPAT pcie_controller

@tbursztyka in that case we also need to update the DTS files to use a different compatible string, and also remove `dts/bindings/pcie/host/intel,pcie.yaml` (which is fine with me - I'm just saying that only changing DT_DRV_COMPAT will not work)

Reply:
hum, yes I did not see this comming. But indeed we need pcie,controller.yaml instead of pcie-controller.yaml
and replace all intel,pcie compat strings to pcie,controller.

So far, all about pcie is generic and intel has nothing specific related to it, so we should be good to go in a pcie,controller generic binding (and this would remove this ugly arch specific ifdef that should not be here)

Reply:
> hum, yes I did not see this coming. But indeed we need pcie,controller.yaml instead of pcie-controller.yaml and replace all intel,pcie compat strings to pcie,controller.

If there's a comma (`,`) present, the first part is expected to be a vendor name, so I'm not sure we can have `pcie` there. Since this is not based on any existing binding, I think we could just use `zephyr` there (like many other Zephyr bindings already do), in which case this would become `zephyr,pcie-controller`. The other alternative would be to leave `pcie-controller.yaml` as-is, since the omission of a vendor and `,` already implies that it's generic.

Reply:
@tbursztyka @najumon1980 let's keep the `pcie-controller.yaml` name as-is for now, since it's not absolutely necessary to rename it at this point (renaming would also require updating the ECAM binding since it includes the pice-controller one.). So just get rid of the intel binding and extend the pcie-controller one.

Reply:
You should remove `intel,pcie.yaml`, as suggested. As for ecam, I thought the compat matching was based on the filename and not an explicitly stated compat string inside the yaml file. I'm not sure if it's possible to extend a binding with a compat string with another (i.e. the compatible in a binding overrides the compatible in a binding that it includes). Did you try this? Do you get errors?

Reply:
..answering myself: it's ok even if the included file has a `compatible: ..` line also. At least I didn't get any errors when I added that to `pice-controller.yaml`.